Bottom: e9b46b0149df07b69b70a643d70c698755f8710f
Top:    d59c995dc8475542e2f8bfc404c254f082a354f0
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-10-01 14:13:02 +0100

Refresh of toporouter-c-re-arrange-the-ma

---

diff --git a/src/toporouter.c b/src/toporouter.c
index 8060b13..224a6b9 100644
--- a/src/toporouter.c
+++ b/src/toporouter.c
@@ -1192,7 +1192,7 @@ vertices_on_line (toporouter_spoint_t *a,
 
   double c, dx;
 
-  if(m == INFINITY || m == -INFINITY) {
+  if (m == INFINITY || m == -INFINITY) {
 
     b0->x = a->x;
     b0->y = a->y + r;
@@ -1214,37 +1214,39 @@ vertices_on_line (toporouter_spoint_t *a,
 }
 
 /**
- * vertices_on_line:
- * Given vertex a, gradient m, and radius r: 
+ * coords_on_line:
+ * Given coordinates ax, ay, gradient m, and radius r:
  *
- * Return vertices on line of a & m at r from a
- */ 
-void
-coords_on_line(gdouble ax, gdouble ay, gdouble m, gdouble r, gdouble *b0x, gdouble *b0y, gdouble *b1x, gdouble *b1y)
+ * Return coordinates on line of a & m at r from a
+ */
+static void
+coords_on_line (double ax, gdouble ay,
+                double m,
+                double r,
+                double *b0x, double *b0y,
+                double *b1x, double *b1y)
 {
+  double c, dx;
 
-  gdouble c, temp;
-  
-  if(m == INFINITY || m == -INFINITY) {
-    *b0y = ay + r;
-    *b1y = ay - r;
+  if (m == INFINITY || m == -INFINITY) {
 
     *b0x = ax;
+    *b0y = ay + r;
+
     *b1x = ax;
-  
+    *b1y = ay - r;
+
     return;
   }
 
-  c = ay - (m * ax);
-
-  temp = sqrt( pow(r, 2) / ( 1 + pow(m, 2) ) );
+  c = ay - m * ax;
+  dx = r / sqrt (1 + pow (m, 2));
 
-  *b0x = ax + temp;
-  *b1x = ax - temp;
-  
-  *b0y = *b0x * m + c;
-  *b1y = *b1x * m + c;
+  *b0x = ax + dx;
+  *b0y = m * *b0x + c;
 
+  *b1x = ax - dx;
+  *b1y = m * *b1x + c;
 }
 
 /**
